api/v1alpha3/awsmachine_types.go

```diff
@@ -28,6 +28,17 @@ const (
 	MachineFinalizer = "awsmachine.infrastructure.cluster.x-k8s.io"
 )
 
+// SecretBackend defines variants for backend secret storage.
+type SecretBackend string
+
+var (
+	// SecretBackendSSMParameterStore defines AWS Systems Manager Parameter Store as the secret backend
+	SecretBackendSSMParameterStore = SecretBackend("ssm-parameter-store")
+
+	// SecretBackendSecretsManager defines AWS Secrets Manager as the secret backend
+	SecretBackendSecretsManager = SecretBackend("secrets-manager")
+)
```

Comment:
Not sure if this was discussed previously, but is there a specific use case in mind for having this as an API field rather than a configuration option on the controller manager?

Is there a specific use case in mind where one would want to choose different backends for individual Clusters/Machines vs having it a global configuration for an instance of the infrastructure provider?

Reply:
Oops, closed it because @Promaethius is around, I'll reply in the other PR,